### DISCUSSION TITLE
Resize IPAM infrastructure

### DISCUSSION THREAD

**Discussion by thelumlaa** [Azure/ipam#92] [Q&A] [answered]
IPAM deployment is a fairly significant investment, especially for smaller companies. Is there any guidance on App Service Plan resizing? Can it be downgraded? What would be the repercussions? Apparently it uses premium tier for diagnostic settings, but we haven't used them once in our few months of using IPAM. Please share your thoughts and experience.

**Comment by DCMattyG** [comment, marked as answer]
Hi @thelumlaa, whilst we haven't tested every possible App Service Plan size, I can tell you the reason we chose the P1v3 as the default is:

- Diagnostic Settings
- Always On
- Private Endpoint support

That said, we have a number of customers who run much smaller instances. Having run the solution in smaller plans as a test, I can tell you that you may see longer start times when restarting the App Service (I've heard of ~30 minutes with the B1 plan), slower API response times (The Engine does quite a bit of computation, and stripping away resources can slow this process down), and possibly API timeouts (many API's are called synchronously, and if using the UI then the slowed response may cause failures).

TL:DR - If you're you're ok with things running a bit slower and losing the Premium features, please feel free to resize to whatever you are comfortable with. We selected the P1v3 as the default because we know it runs well and supports the advanced features many customers are looking for, but we fully recognize that this may not be the ideal size/cost for all users. My advice would be to slowly step it down to your target size, and do some testing to make sure everything is still operating within an acceptable threshold for you.
